src/reducers/existingThoughtChange.js

```diff
@@ -66,37 +65,58 @@ export default (state, { oldValue, newValue, context, showContexts, thoughtsRank
   const oldPath = rankThoughtsFirstMatch(thoughtsOld, { state })
   const newPath = oldPath.slice(0, oldPath.length - 1).concat({ value: newValue, rank: oldPath.slice(oldPath.length - 1)[0].rank })
 
-  /**
-      .removing if the old path is already in the object
-      .Checking for all the descendants and updating their path too
-      .limiting number of recenlty edited thoughts
-      .merging the paths with common majority subcontext
-      .concating recently edited thought path only if no merge has happened
-      .sorting by last updated
-  */
-
   let isMerged = false // eslint-disable-line fp/no-let
 
-  // if we dont use isMerged variable we have have to iterate the array one more time to find if any merge has happened
+  const newPathIndex = pathToIndex(newPath)
+  const oldPathIndex = pathToIndex(oldPath)
 
-  const recentlyEdited = reverse(sortBy(
-    [...state.recentlyEdited]
-      .filter(recentlyEditedThought => !equalPath(recentlyEditedThought.path, oldPath))
-      .map(recentlyEditedThought => subsetThoughts(recentlyEditedThought.path, oldPath) ? Object.assign({}, recentlyEditedThought, { path: newPath.concat(recentlyEditedThought.path.slice(newPath.length)) }) : recentlyEditedThought)
-      .slice(0, RECENTLY_EDITED_THOUGHTS_LIMIT)
-      .map(recentlyEditedThought => {
-        // checking for availability of majoirty subcontext between two rankedThoughts[]
-        const subcontextIndex = checkIfPathShareSubcontext(recentlyEditedThought.path, newPath)
-        const timeDiffInSec = timeDifference(timestamp(), recentlyEditedThought.lastUpdated)
+  const updatedRecentlyEdited = { ...state.recentlyEdited }
 
-        // this variable sets to true when there is atleast one recently edited thought it can merge to
-        if (subcontextIndex > -1 && timeDiffInSec <= 7200) isMerged = true // eslint-disable-line fp/no-mutating-methods
+  if (updatedRecentlyEdited[oldPathIndex]) {
+    delete updatedRecentlyEdited[oldPathIndex] // eslint-disable-line fp/no-delete
```

Comment:
Please do not use delete in arrays, it will leave holes that might be so hard to debug. 
`updatedRecentlyEdited[oldPathIndex]` would have a value of undefined which I really don't think is good. 
`updatedRecentlyEdited = updatedRecentlyEdited.filter((updatedValue, index)=>index !== oldPathIndex)` would give you a refined array instead

Reply:
https://rules.sonarsource.com/javascript/RSPEC-2870

Reply:
@Ogutu-Brian `updatedRecentlyEdited` is an object, not an array.

Reply:
Thanks @raineorshine ,
I didn't know that.